docs/ref/contrib/messages.txt

```diff
@@ -214,24 +222,28 @@ is a mapping of the message level names to their numeric value::
     </ul>
     {% endif %}
 
+**Outside of templates**, you can use also use ``request.messages``::
 
-
-**Outside of templates**, you can use
-:func:`~django.contrib.messages.get_messages`::
-
-    from django.contrib.messages import get_messages
-
-    storage = get_messages(request)
-    for message in storage:
+    for message in request.messages::
         do_something_with_the_message(message)
 
 For instance, you can fetch all the messages to return them in a
 :ref:`JSONResponseMixin <jsonresponsemixin-example>` instead of a
 :class:`~django.views.generic.base.TemplateResponseMixin`.
 
+.. function:: get_messages(request)
+
 :func:`~django.contrib.messages.get_messages` will return an
-instance of the configured storage backend.
+instance of the configured storage backend, or an empty list
+if the middleware is not installed.
```

Comment:
Hah. I added that sentence because I don't like removing features that aren't hard to support. :).

Basically: `get_messages = lambda r: r.messages if hasattr(r, 'messages') else []`